src/rules/pylint/rules/too_many_return_statements.rs

```diff
@@ -0,0 +1,266 @@
+use crate::ast::helpers::identifier_range;
+use crate::define_violation;
+use crate::registry::Diagnostic;
+use crate::source_code::Locator;
+use crate::violation::Violation;
+
+use ruff_macros::derive_message_formats;
+use rustpython_ast::{ExcepthandlerKind, Stmt, StmtKind};
+
+define_violation!(
+    pub struct TooManyReturnStatements {
+        pub returns: usize,
+        pub max_returns: usize,
+    }
+);
+impl Violation for TooManyReturnStatements {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let TooManyReturnStatements {
+            returns,
+            max_returns,
+        } = self;
+        format!("Too many return statements ({returns}/{max_returns})")
+    }
+}
+
+fn num_returns(stmts: &[Stmt]) -> usize {
```

Comment:
Could we instead reuse `ReturnStatementVisitor` here, from `src/rules/flake8_annotations/rules.rs`?